inference-engine/src/transformations/include/ngraph_ops/topk_ie.hpp

```diff
@@ -19,24 +20,28 @@ class INFERENCE_ENGINE_API_CLASS(TopKIE) : public Op {
     static constexpr NodeTypeInfo type_info{"TopKIE", 1};
     const NodeTypeInfo& get_type_info() const override { return type_info; }
 
-    TopKIE(const Output<Node> &data,
-           const Output<Node> &k,
+    TopKIE(const Output<Node>& data,
+           const Output<Node>& k,
            const int64_t axis,
-           const std::string& mode,
-           const std::string& sort,
-           const Shape& output_shape);
+           const ngraph::op::TopKMode mode,
+           const ngraph::op::TopKSortType sort,
+           const element::Type& index_element_type = element::i32);
 
     void validate_and_infer_types() override;
 
-    std::shared_ptr<Node> copy_with_new_args(const NodeVector& new_args) const override;
-    int64_t get_axis();
-    std::string get_mode();
-    std::string get_sort_type();
-    Shape get_output_shape();
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override;
 
-    int64_t axis;
-    std::string mode, sort_type;
-    Shape output_shape;
+    int64_t get_axis() { return m_axis;}
+
+    ngraph::op::TopKMode get_mode() { return m_mode; }
+
+    ngraph::op::TopKSortType get_sort_type() { return m_sort_type; }
+
+private:
+    int64_t m_axis;
+    ngraph::op::TopKMode m_mode;
+    ngraph::op::TopKSortType m_sort_type;
+    element::Type m_index_element_type;
```

Comment:
We do not pass `m_index_element_type` to IE and resolve it for TopK-3 earlier. Do we really need this attribute?

Reply:
Actually v1::TopK and v3::TopK have `m_index_element_type` attribute. But I don't want to change old logic in this PR. Let's discuss it offline and I'll make a change if it needed.

Reply:
Finally decided to remove m_index_element_type from TopKIE because we handle index_element_type for v1::TopK and v3::TopK in convert_topk3 transformation.